homeassistant/components/axis/config_flow.py

```diff
@@ -0,0 +1,202 @@
+"""Config flow to configure Axis devices."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_DEVICE, CONF_HOST, CONF_MAC, CONF_NAME, CONF_PASSWORD, CONF_PORT,
+    CONF_USERNAME)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.util.json import load_json
+
+from .const import CONF_MODEL, DOMAIN
+from .device import get_device
+from .errors import AlreadyConfigured, AuthenticationRequired, CannotConnect
+
+CONFIG_FILE = 'axis.conf'
+
+EVENT_TYPES = ['motion', 'vmd3', 'pir', 'sound',
+               'daynight', 'tampering', 'input']
+
+PLATFORMS = ['camera']
+
+AXIS_INCLUDE = EVENT_TYPES + PLATFORMS
+
+AXIS_DEFAULT_HOST = '192.168.0.90'
+AXIS_DEFAULT_USERNAME = 'root'
+AXIS_DEFAULT_PASSWORD = 'pass'
+DEFAULT_PORT = 80
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_HOST, default=AXIS_DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_USERNAME, default=AXIS_DEFAULT_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD, default=AXIS_DEFAULT_PASSWORD): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+}, extra=vol.ALLOW_EXTRA)
+
+
+@callback
+def configured_devices(hass):
+    """Return a set of the configured devices."""
+    return set(entry.data[CONF_DEVICE][CONF_HOST] for entry
+               in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class AxisFlowHandler(config_entries.ConfigFlow):
+    """Handle a Axis config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Axis config flow."""
+        self.device_config = {}
+        self.model = None
+        self.name = None
+        self.serial_number = None
+
+        self.discovery_schema = {}
+        self.import_schema = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a Axis config flow start.
+
+        Manage device specific parameters.
+        """
+        from axis.vapix import VAPIX_MODEL_ID, VAPIX_SERIAL_NUMBER
+        errors = {}
+
+        if user_input is not None:
+            try:
+                if user_input[CONF_HOST] in configured_devices(self.hass):
```

Comment:
A host can change, since you also have access to a serial number, should you use that to remove the old one when an entry is found at a new location?

Reply:
Sure, I'll have a look at this in a separate PR, just so I can finish this one :)